inspirehep/modules/records/receivers.py

```diff
@@ -286,11 +286,28 @@ def earliest_date(sender, json, *args, **kwargs):
         json['earliest_date'] = earliest_date
 
 
+def generate_texkey(record):
+    return 'ciccio'
+
+
+def insert_texkeys(record, new_texkey):
+    record['texkeys'] = [new_texkey]
+
+
+def update_texkey(record, new_texkey):
```

Comment:
do you really need two different methods here for insertion and updating?
if you do simply

```ipython
record.setdefault('texkeys', [])
record['texkeys'].insert(0, new_texkey)
```
it would work for both insert and update.